### PR TITLE
Initial commit UniqueLengthBatchWindow

### PR DESCRIPTION
**Pull request by hmrajas**

### CODE REVIEW

File: src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            this.uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
```
Comment (dilini-muthumala) [inline src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java]: We need to check whether `attributeExpressionExecutors[0] instanceof VariableExpressionExecutor`

File: src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    this.windowLength = (Integer)
+                            (((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Length should be an INT");
```
Comment (dilini-muthumala) [inline src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java]: Let's try to give a more helpful error message to the user. How about something like `"Unique Length Batch window's 'length' parameter should be INT, but found " + attributeExpressionExecutors[1].getReturnType()` ?

File: src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Length Batch window should have constant " +
+                        "for length parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+        } else if (attributeExpressionExecutors.length == 3) {
+            this.uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
```
Comment (dilini-muthumala) [inline src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java]: Do the instanceof check

File: src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    this.windowLength = (Integer)
+                            (((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Length should be an INT");
```
Comment (dilini-muthumala) [inline src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java]: As explained above, let's try to improve this error message as well.

File: src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+                throw new ExecutionPlanValidationException("Unique Length Batch window should have constant " +
+                        "for length parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            if (attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.BOOL) {
+                    this.isFirstUniqueEnabled = (boolean) (((ConstantExpressionExecutor)
+                            attributeExpressionExecutors[2]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Third Value should be boolean value");
```
Comment (dilini-muthumala) [inline src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java]: This should be "Third parameter". Let's try to improve this error message as well, as explained above.

File: src/test/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowTestCase.java [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+        inputHandler.send(new Object[]{"IBM", 10f, 0});
+        inputHandler.send(new Object[]{"IBM", 20f, 1});
+        inputHandler.send(new Object[]{"IBM", 30f, 0});
+        inputHandler.send(new Object[]{"IBM", 40f, 1});
+        inputHandler.send(new Object[]{"IBM", 50f, 0});
+        inputHandler.send(new Object[]{"IBM", 60f, 1});
+        Thread.sleep(500);
+        executionPlanRuntime.shutdown();
+//        Assert.assertEquals(0, inEventCount);
+//        Assert.assertTrue(eventArrived);
```
Comment (thiliA) [inline src/test/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowTestCase.java]: @hmrajas : Is there any particular reason to comment out the above assertions? Same applies to line 263,265-267.

**Pull request merged**